tests/unit/test_lambda.py

```diff
@@ -85,6 +94,124 @@ def test_list_non_existant_function_versions_returns_error(self):
             self.assertEqual(self.RESOURCENOTFOUND_MESSAGE % lambda_api.func_arn(self.FUNCTION_NAME),
                              result['message'])
 
+    def test_create_alias(self):
+        with self.app.test_request_context():
+            self._create_function(self.FUNCTION_NAME)
+            lambda_api.publish_version(self.FUNCTION_NAME)
+
+            flask.request.data = json.dumps({
+                "Name": self.ALIAS_NAME,
+                "FunctionVersion": '1',
+                "Description": ''
+            })
+            result = json.loads(lambda_api.create_alias(self.FUNCTION_NAME).get_data())
+
+            expected_result = {u'AliasArn': lambda_api.func_arn(self.FUNCTION_NAME) + ":" + self.ALIAS_NAME,
+                               u'FunctionVersion': '1', u'Description': '', u'Name': self.ALIAS_NAME}
+            self.assertDictEqual(expected_result, result)
+
+    def test_create_alias_on_non_existant_function_returns_error(self):
+        with self.app.test_request_context():
+            result = json.loads(lambda_api.create_alias(self.FUNCTION_NAME).get_data())
+            self.assertEqual(self.RESOURCENOTFOUND_EXCEPTION, result['__type'])
+            self.assertEqual(self.RESOURCENOTFOUND_MESSAGE % lambda_api.func_arn(self.FUNCTION_NAME),
+                             result['message'])
+
+    def test_create_alias_returns_error_if_already_exists(self):
+        with self.app.test_request_context():
+            self._create_function(self.FUNCTION_NAME)
+            lambda_api.publish_version(self.FUNCTION_NAME)
+            flask.request.data = json.dumps({
```

Comment:
I'm wondering whether we should have this as an integration tests, rather than a unit test. I think using `test_request_context()` and *reading* from `flask.request` is generally fine (and a recommended pattern, or at least a ["testing trick"](http://flask.pocoo.org/docs/0.12/testing/#other-testing-tricks) ), but manually *modifying* `flask.request` to invoke a certain code path that is usually triggered by a REST call seems like a bit of an anti-pattern.

One of the main problems I see with the flask unit testing approach is that LocalStack makes heavy use of proxies and invocation listeners (not the Lambda API currently, but other services are using it). So, if our testing is based on mocking flask requests, we might be potentially be missing (or skipping) parts of the service invocation chain. I'd rather not establish this as a pattern. 

I don't think it makes much sense to re-write the tests in this PR at this point (we can go ahead and merge as-is)
but I feel like this is something we should consider in the future.

Reply:
I agree. Also, reading the Flask doc. The proper way to test a request with body data is like this 
```
self.app.post('/login', data=dict(
        username=username,
        password=password
    ), follow_redirects=True)
```
It would be cleaner than setting the flask request directly. I will do it.